### PR TITLE
Update to 1.19.4

### PR DESCRIPTION
**Pull request by TheLimeGlass**
### Description
Updates to 1.19.4.
---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/entity/SimpleEntityData.java [hunk 1]
```diff
@@ -23,114 +23,7 @@
...
-import org.bukkit.entity.Warden;
-import org.bukkit.entity.WaterMob;
-import org.bukkit.entity.Witch;
-import org.bukkit.entity.Wither;
-import org.bukkit.entity.WitherSkeleton;
-import org.bukkit.entity.WitherSkull;
-import org.bukkit.entity.Zoglin;
-import org.bukkit.entity.Zombie;
-import org.bukkit.entity.ZombieHorse;
+import org.bukkit.entity.*;
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/entity/SimpleEntityData.java:26]: Don't know how this happened, I don't even have condense imports enabled.

Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/entity/SimpleEntityData.java:26]: Oh dear! this is a big problem, we will have to close the issue for that! Sorry.

**Comment by TheLimeGlass** [comment]
`java.lang.NoClassDefFoundError: org/bukkit/entity/Boat$Type` well that's not good 1.13...

**Comment by bluelhf** [comment]
Should we remove 1.19.3 test environment in favour of 1.19.4?

**Comment by TheLimeGlass** [comment]
> Should we remove the 1.19.3 test environment in favour of 1.19.4?

It's a pretty big change in versions now since 1.19.3 Mojang is doing this new release schedule with these features, so we're keeping it for now. Maybe atleast 1 minor version behind.

File: src/main/java/ch/njol/skript/entity/SimpleEntityData.java (hunk 1 above)
Comment (Pikachu920) [reply src/main/java/ch/njol/skript/entity/SimpleEntityData.java:26]: > Don't know how this happened, I don't even have condense imports enabled.

if you're using intellij I've noticed you often have to disable it AND change the minimum number of imports to 9999 or something.

**Comment by TheLimeGlass** [comment]
Reverted the boat entity data changes. They're not critical to this update. Boat.Type is not present in 1.17 and lower, so it'll need some serious refactoring. The deprecated methods still operate fine, and Skript doesn't even fully support the new boat types.

File: src/main/resources/lang/default.lang [hunk 2]
```diff
@@ -491,6 +493,15 @@ entities:
 	dark oak boat:
 		name: dark oak boat¦s
 		pattern: dark oak boat(|1¦s)
+	mangrove boat:
+		name: mangrove boat¦s
+		pattern: mangrove boat(|1¦s)
+	bamboo boat:
```
Comment (Moderocky) [inline src/main/resources/lang/default.lang:499]: I know this is referred to as 'bamboo raft' in some places - do we need both variations here?

Comment (TheLimeGlass) [reply src/main/resources/lang/default.lang:499]: For the aliases yes. The id is minecraft:bamboo_raft
This does not matter what the name is. This is used by the EntityData class.

I will update it to raft though.

Comment (TheLimeGlass) [reply src/main/resources/lang/default.lang:499]: Again recoding the EntityData for BoatData eventually. Currently Bamboo and Cherry boats aren't released, as it's 1.17+ as mentioned above.

**Comment by GStudiosX2** [comment]
will their be like expressions/effects for like setting scale of display and everything that can be done with displays ?

**Comment by TheLimeGlass** [comment]
> will their be like expressions/effects for like setting scale of display and everything that can be done with displays ?

Not yet. This just allows Skript to understand what they are.

**Comment by GStudiosX2** [comment]
> > will their be like expressions/effects for like setting scale of display and everything that can be done with displays ?
> 
> Not yet. This just allows Skript to understand what they are.

will their eventually be for 2.7.0 or 2.7.0-beta2 ? or will it only be coming to next release 2.8 ?

**Comment by TheLimeGlass** [comment]
It has a 2.7 tag.

**Comment by GStudiosX2** [comment]
> It has a 2.7 tag.

i mean like for expressions/effect to interact with the new display entities

**Comment by DelayedGaming** [comment]
> > It has a 2.7 tag.
> 
> i mean like for expressions/effect to interact with the new display entities

It shall be added in the next coming release whenever a developer or a contributor has integrated the feature into Skript.

**Review by Moderocky — APPROVED** [review]

**Review by AyhamAl-Ali — APPROVED** [review]

**Pull request merged**